app/knights/knight.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+    def prepare(
+            self,
+            armour: list[dict],
+            weapon: dict,
+            potion: dict
+    ) -> None:
+        if armour is not None:
+            for item in armour:
+                self.protection += item["protection"]
+
+        self.power += weapon["power"]
+
+        if potion is not None and "effect" in potion:
+            effects = potion["effect"]
+            for effect_type, value in effects.items():
+                if effect_type == "power":
+                    self.power += value
+                elif effect_type == "hp":
+                    self.hp += value
+                elif effect_type == "protection":
+                    self.protection += value
```

Comment:
Just as an alternative implementation, you could use this approach to remove the `if elif` conditions:
```suggestion
            for effect_type, value in effects.items():
                effect_value = getattr(self, effect_type)
                effect_value += value
                setattr(self, effect_type, effect_value)
```